File: 4_Clojure_Examples/2_Shapes_LSP/square.clj
(ns square)

(defn make-square [x y side-length]
	(make-rectangle x y side-length side-length))

(defn draw-square [square color]
	(println "This is a " color " square at pos " (:x square) "," (:y square) " with side length " (:side-length square)))

über defmulti und defmethod?
		
	def set_side_lengths horizontal_length, vertical_length
		if horizontal_length != vertical_length
			 throw Exception.new("For squares, both sides have to have equal lengths.")
		end
		super.set_side_lengths(horizontal_length, vertical_length)
	end
	
end

File: 1_Java_Examples/2_Shapes_LSP/Square.java
public class Square extends Rectangle {
	
	public Square(int x, int y, int sideLength) {
		super(x, y, sideLength, sideLength);
	}
	

	@Override public void setSideLengths(int horizontalLength, int verticalLength) {
		if (horizontalLength != verticalLength) throw new IllegalArgumentException("For squares, both sides have to have equal lengths.");
		super.setSideLengths(horizontalLength, verticalLength);
	}
	
}